Documents/Automation/EncryptRootVolume/Documents/npark-encryptrootvolume.json

```diff
@@ -121,41 +131,50 @@
       "inputs": {
         "Service": "ec2",
         "Api": "CopySnapshot",
-        "SourceSnapshotId": "{{extractSnapshotID.SNAPSHOTID}}",
+        "SourceSnapshotId": "{{extractSnapshotId.SnapshotId}}",
         "SourceRegion": "{{global:REGION}}",
         "Encrypted": true,
-        "KmsKeyId": "{{KmsKeyId}}",
+        "KmsKeyId": "{{kmsKeyId}}",
         "DestinationRegion": "{{global:REGION}}"
       },
       "outputs": [
         {
-          "Name": "EncryptedSnapshotID",
+          "Name": "encryptedSnapshotId",
           "Selector": "$.SnapshotId",
           "Type": "String"
         }
       ],
-      "name": "CopySnapshot",
+      "name": "copyAndEncryptSnapshot",
       "action": "aws:executeAwsApi",
       "timeoutSeconds": 3600,
       "onFailure": "Abort"
     },
     {
+      "name": "waitForSnapshot",
```

Comment:
You haven't specified the onFailure behavior here. My expectation is that you'll delete the snapshot you have created.
See my overall review comment for a branch analysis.

Reply:
corrected in the latest commit